compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1150,7 +1150,7 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
           // classes defined in a such arguments should not be entered into the
           // enclosing class.
           c.exprContext(mdef, exprOwner)
-        case None => c
```

Comment:
This is a perfect example of why we shouldn't turn off exhaustiveness warnings when guards are used ;) (cf: https://github.com/scala/scala/pull/5617)